lua/better-breachable-doors/sv_breach.lua

```diff
@@ -0,0 +1,757 @@
+            local otherDoorName = string.Trim( otherDoor:GetName() )
+            local doorsHaveSameName = otherDoorName ~= "" and otherDoorName == doorName and otherDoor ~= door
+            local doorsHaveSlaveNameConnection = otherDoor:GetInternalVariable( "slavename" ) == doorName
+
+            if doorsHaveSameName or doorsHaveSlaveNameConnection then
+                result[ #result + 1 ] = otherDoor
+            end
+        end
+    end
+
+    -- Remove the original door from the list
+    table.RemoveByValue( result, door )
+
+    BBD.ConnectedDoors[ door ] = result
+    return result
+end
+
+-- Returns whether or not the given door has any connected doors.
+-- Uses cached connections data if available.
+---@param door BBD.Door Door to check
+---@return boolean # Whether or not the door has any connected doors
+BBD.DoorHasConnections = function( door )
+    local connectedDoors = BBD.GetConnectedDoors( door )
+    return connectedDoors ~= nil and #connectedDoors > 0
+end
+
+--#endregion Door Connection Functions
+
+--#region AngularMove Port
+
+-- Based on `CBaseEntity::GetMoveDoneTime`
+---@param ent Entity Entity to get the move done time for
+---@return number # The time, in seconds, it will take to move the door or -1 if an error occurred
+local function GetMoveDoneTime( ent )
+    local moveDoneTime = ent:GetInternalVariable( "m_flMoveDoneTime" )
+    return moveDoneTime >= 0 and moveDoneTime - ent:GetInternalVariable( "ltime" ) or -1
+end
+
+-- Based on `CBaseEntity::WillSimulateGamePhysics`
+---@param ent Entity Entity to check
+---@return boolean # Whether or not the entity will simulate game physics
+local function WillSimulateGamePhysics( ent )
+    if not ent:IsPlayer() then
+        local moveType = ent:GetMoveType()
+
+        if moveType == MOVETYPE_NONE or moveType == MOVETYPE_VPHYSICS then
+            return false
+        end
+
+        if SERVER then
+            -- MOVETYPE_PUSH is only valid on the server
+            if moveType == MOVETYPE_PUSH and GetMoveDoneTime( ent ) <= 0 then
+                return false
+            end
+        end
+    end
+
+    return true
+end
+
+-- Based on `CBaseEntity::CheckHasGamePhysicsSimulation`
+---@param ent Entity Door to check
+local function CheckHasGamePhysicsSimulation( ent )
+    if not IsValid( ent ) then return end
+
+    local isSimulating = WillSimulateGamePhysics( ent )
+
+    if isSimulating ~= ent:IsEFlagSet( EFL_NO_GAME_PHYSICS_SIMULATION ) then
+        return
+    end
+
+    if isSimulating then
+        ent:RemoveEFlags( EFL_NO_GAME_PHYSICS_SIMULATION )
+    else
+        ent:AddEFlags( EFL_NO_GAME_PHYSICS_SIMULATION )
+    end
+end
+
+-- Based on `CBaseEntity::SetMoveDoneTime`
+---@param ent Entity Door to set the move done time for
+---@param delay number Time, in seconds, to set the move done time to
+local function SetMoveDoneTime( ent, delay )
+    if not IsValid( ent ) then return end
+    if not delay then return end
```

Comment:
Can `delay` be `nil`? If so, annotate it, and perform that check before the `IsValid` check

```suggestion
---@param delay number? Time, in seconds, to set the move done time to
local function SetMoveDoneTime( ent, delay )
    if not delay then return end
    if not IsValid( ent ) then return end
```